### DISCUSSION TITLE
[Zephyr CMake] How use zephyr_library_link_libraries() with PRIVATE keywords?

### DISCUSSION THREAD

**Discussion by Nulllix** [zephyrproject-rtos/zephyr#83068] [Q&A] [answered]
Hi, I would like to link other libraries with my zephyr library using the PRIVATE keyword and I would like to use the `zephyr_library_link_libraries()` function for this. But you cannot specify PRIVATE/PUBLIC/INTERFACE keywords in the `zephyr_library_link_libraries()` function, it always links libraries using the PUBLIC keyword:
```
function(zephyr_library_link_libraries item)
  target_link_libraries(${ZEPHYR_CURRENT_LIBRARY} PUBLIC ${item} ${ARGN})
endfunction()
```
Can you tell me the reason why this function has such an implementation? And how can I link other libraries with my zephyr library using the PRIVATE keyword?

**Comment by Nulllix** [comment]
Up discussions 😊

**Comment by yashi** [comment, marked as answer]
```cmake
zephyr_library_link_libraries(PRIVATE your_item)
```

**Reply by Nulllix** [reply]
Oh... sorry. My mistake. I didn't think about the fact that ${ARGN} could also pass the PRIVATE/INTERFACE keywords :neutral_face:
